Hazel/src/Hazel/Core/Assert.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#ifdef HZ_ENABLE_ASSERTS
+
+	namespace Hazel::Assert
+	{
+		// Returns the simple file name rather than full path as suggested by LovelySanta
+		constexpr const char* CurrentFileName(const char* path) {
+				const char* file = path;
+				while (*path)
+				{
+					if (*path == '/' || *path == '\\')
+						file = ++path;
+					else
+						path++;
+				}
+				return file;
+			}
+	}
+
+	// Alteratively we could use the same "default" message for both "WITH_MSG" and "NO_MSG" and
+	// provide support for custom formatting by concatenating the formatting string instead of having the format inside the default message
+	#define HZ_INTERNAL_ASSERT_IMPL(type, check, msg, ...) { if(!(check)) { HZ##type##ERROR(msg, __VA_ARGS__); HZ_DEBUGBREAK(); } }
+	#define HZ_INTERNAL_ASSERT_WITH_MSG(type, check, ...) HZ_INTERNAL_ASSERT_IMPL(type, check, "Assertion '{0}' failed: {1}", HZ_STRINGIFY_MACRO(check), __VA_ARGS__)
```

Comment:
Having a closer look, the message should explain the assert, so in this case the check should not be part of the message. It would become too long otherwise...
```suggestion
	#define HZ_INTERNAL_ASSERT_WITH_MSG(type, check, ...) HZ_INTERNAL_ASSERT_IMPL(type, check, "Assertion failed: {0}", __VA_ARGS__)
```